modules/datastore/src/Plugin/DatasetInfo/DatastoreInfo.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\datastore\Plugin\DatasetInfo;
+
+use Drupal\common\DatasetInfoPluginBase;
+use Drupal\datastore\DatastoreService;
+use Drupal\datastore\Service\Info\ImportInfo;
+use Drupal\datastore\Service\ResourceLocalizer;
+use Drupal\metastore\ResourceMapper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the dataset_info.
+ *
+ * @DatasetInfoPlugin(
+ *   id = "datastore_info",
+ * )
+ */
+class DatastoreInfo extends DatasetInfoPluginBase {
```

Comment:
Q: Should this class be called DatasetInfo?

Reply:
I don't think so, I think it should be close to the plugin ID. Maybe DatastoreDatasetInfo would be more precise but that seems kind of silly.